usr/share/rear/lib/layout-functions.sh

```diff
@@ -992,4 +992,180 @@ function apply_layout_mappings() {
     is_true $apply_layout_mappings_succeeded && return 0 || return 1
 }
 
+# Keeps track of last partition created for a specific disk
+# Associative array indexed by disk
+# e.g. last_partition_number[/dev/sda]=1
+declare -A last_partition_number=()
```

Comment:
Associative arrays are not supported with bash 3.x
and ReaR must work with bash 3.x, see
https://github.com/rear/rear/wiki/Coding-Style

Reply:
OK, but who still has bash 3.x ???

Reply:
SLES11 has bash 3.x and SLES11 is still supported by ReaR
and what about older RHEL that are still supported by ReaR:
https://github.com/rear/rear.github.com/blob/master/documentation/release-notes-2-5.md
```
Supported and Unsupported Operating Systems
...
ReaR... is supported on the following Linux based operating systems:
...
RHEL 5, 6 and 7
...
SLES 11, 12 and 15
```

Reply:
Alright. Will rework the code.